src/xcoobee/api/EventSubscriptionsApi.js

```diff
@@ -151,75 +88,120 @@ const deleteEventSubscription = (apiUrlRoot, apiAccessToken, arrayOfEventNames,
  *
  * @throws {XcooBeeError}
  */
-const listEventSubscriptions = (apiUrlRoot, apiAccessToken, campaignId) => {
-  ApiUtils.assertAppearsToBeACampaignId(campaignId);
-  const query = `
-    query listEventSubscriptions($campaignId: String!) {
-      event_subscriptions(campaign_cursor: $campaignId) {
-        data {
-          campaign_cursor
-          date_c
-          event_type
-          handler
-          owner_cursor
-        }
-        page_info {
-          end_cursor
-          has_next_page
-        }
+const listEventSubscriptions = (apiUrlRoot, apiAccessToken, referenceId, referenceType) => {
+  const query = `{
+    event_subscriptions {
+      data {
+        owner_cursor
+        reference_cursor
+        reference_type
+        topic
+        channel
+        handler
+        date_c
       }
     }
-  `;
+  }`;
+
   return ApiUtils.createClient(apiUrlRoot, apiAccessToken).request(query, {
-    campaignId,
+    referenceId,
+    referenceType,
   })
-    .then((response) => {
-      const { event_subscriptions } = response;
+    .then((response) => response.event_subscriptions)
+    .catch((err) => {
+      throw ApiUtils.transformError(err);
+    });
+};
 
-      // TODO: Remove the following once `event_subscriptions` returns a valid `page_info`
-      // property or no longer does cursor-based pagination.
-      event_subscriptions.page_info = event_subscriptions.page_info || { end_cursor: null, has_next_page: false };
+/**
+ * Fetches available subscriptions and channels for the user
+ *
+ * @async
+ * @param {string} apiUrlRoot - The root of the API URL.
+ * @param {ApiAccessToken} apiAccessToken - A valid API access token.
+ * @param {string} referenceId - id of related enyity (i.e. campaignId)
+ * @param {string} referenceType - type of related entity (i.e. campaign, funding_panel)
+ *
+ * @returns {Promise<Array>} - All available subscriptions to subscribe.
+ *
+ * @throws {XcooBeeError}
+ */
+const getAvailableSubscriptions = (apiUrlRoot, apiAccessToken, referenceId, referenceType) => {
+  const query = `query getAvailableSubscriptions($referenceType: EventReferenceType, $referenceId: String){
+    available_subscriptions (reference_type: $referenceType, reference_cursor: $referenceId) {
+        topic
+        channels
+    }
+  }`;
 
-      // TODO: return valid event type
+  return ApiUtils.createClient(apiUrlRoot, apiAccessToken).request(query, {
+    referenceId,
+    referenceType,
+  })
+    .then((response) => response.available_subscriptions)
+    .catch((err) => {
+      throw ApiUtils.transformError(err);
+    });
+};
 
-      return event_subscriptions;
-    })
+/**
+ * Fetches available subscriptions and channels for the user
+ *
+ * @async
+ * @param {string} apiUrlRoot - The root of the API URL.
+ * @param {ApiAccessToken} apiAccessToken - A valid API access token.
+ * @param {string} topic - topic of subscription
+ * @param {string} channel - channel of subscription

```

Comment:
Please end param descriptions with correct punctuation (probably a period in all cases).
Please start sentences with a capital letter. Thanks.